src/Analyzers/CSharp/CodeFixes/RemoveUnnecessaryDiscardDesignation/RemoveUnnecessaryDiscardDesignationCodeFixProvider.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.RemoveUnnecessaryDiscardDesignation
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp,
+        Name = nameof(RemoveUnnecessaryDiscardDesignationCodeFixProvider)), Shared]
+    internal partial class RemoveUnnecessaryDiscardDesignationCodeFixProvider : SyntaxEditorBasedCodeFixProvider
```

Comment:
Consider renaming to `CSharpRemoveUnnecessaryDiscardDesignationCodeFixProvider` for consistency - analyzers/fixers in language-specific layer have the language prefix.

Reply:
done.